src/main/java/uk/gov/digital/ho/hocs/queue/complaints/ComplaintService.java

```diff
@@ -28,67 +31,109 @@ public class ComplaintService {
+    }
 
-        log.info("createComplaint, started : type {}", complaintData.getComplaintType());
-
-        String untrustedS3ObjectName = documentS3Client.storeUntrustedDocument(ORIGINAL_FILENAME, complaintData.getFormattedDocument());
-
-        log.info("Untrusted document stored in {}, for messageId: {}", untrustedS3ObjectName, clientContext.getCorrelationId());
-
-        DocumentSummary documentSummary = new DocumentSummary(ORIGINAL_FILENAME, DOCUMENT_TYPE, untrustedS3ObjectName);
-
-        var createRequest = composeCreateCaseRequest(complaintData, complaintTypeData, documentSummary);
-        CreateCaseResponse createCaseResponse = workflowClient.createCase(createRequest);
-
-        UUID caseUUID = createCaseResponse.getUuid();
-
-        log.info("createComplaint, create case : caseUUID : {}, reference : {}", caseUUID, createCaseResponse.getReference());
-
-        /*
-            We can swallow exceptions from this point onward, because we would prefer to reduce duplicates. And by this
-            point we have the document stored, and the case is created.
-         */
+    private DocumentSummary createDocument(ComplaintData complaintData) {
         try {
-            UUID stageForCaseUUID = caseworkClient.getStageForCase(caseUUID);
-            log.info("createComplaint, get stage for case : caseUUID : {}, stageForCaseUUID : {}", caseUUID, stageForCaseUUID);
+            String untrustedS3ObjectName = documentS3Client.storeUntrustedDocument(ORIGINAL_FILENAME, complaintData.getFormattedDocument());
+            log.info("Untrusted document stored in {}, for messageId: {}", untrustedS3ObjectName, clientContext.getCorrelationId());
+            messageLogService.updateMessageLogEntryStatus(clientContext.getCorrelationId(), Status.CASE_DOCUMENT_CREATED);
+            return new DocumentSummary(ORIGINAL_FILENAME, DOCUMENT_TYPE, untrustedS3ObjectName);
+        } catch (Exception e) {
+            messageLogService.updateMessageLogEntryStatus(clientContext.getCorrelationId(), Status.CASE_DOCUMENT_FAILED);
+            throw new ApplicationExceptions.DocumentCreationException(e.getMessage(), LogEvent.CASE_DOCUMENT_CREATION_FAILURE);
+        }
+    }
 
-            caseworkClient.updateStageUser(caseUUID, stageForCaseUUID, UUID.fromString(clientContext.getUserId()));
+    private UUID createCase(ComplaintData complaintData, ComplaintTypeData complaintTypeData, DocumentSummary documentSummary) {
+        try {
+            var createRequest = composeCreateCaseRequest(complaintData, complaintTypeData, documentSummary);
+            var createCaseResponse = workflowClient.createCase(createRequest);
+            log.info("createComplaint, create case : caseUUID : {}, reference : {}", createCaseResponse.getUuid(), createCaseResponse.getReference());
+            messageLogService.updateMessageLogEntryStatus(clientContext.getCorrelationId(), Status.CASE_CREATED);
+            return createCaseResponse.getUuid();
```

Comment:
[OOS] - we could look at this returning the stage as well if one is currently present on creation.